### PR TITLE
ENH: filter for PySpark backend

### PR DESCRIPTION
**Pull request by hjoo**
Introduces predicate handling in `ops.Selection` for `filter()` functionality.

Also implemented a few missing comparison operations:

- `And`
- `Or`
- `NotEquals`
- `Less`
- `LessEquals`

### CODE REVIEW

File: ibis/pyspark/compiler.py [hunk 1]
```diff
@@ -71,8 +71,8 @@ def compile_selection(t, expr, scope, **kwargs):
     # Cache compile results for tables
     op = expr.op()
 
-    # TODO: Support predicates and sort_keys
-    if op.predicates or op.sort_keys:
+    # TODO: Support sort_keys
```
Comment (toryhaavik) [inline ibis/pyspark/compiler.py]: could you please file an issue and link it here? i worry we lose track of TODOs if we step away for too long

Comment (hjoo) [reply ibis/pyspark/compiler.py]: Yep, could you link me to the JIRA project?

Comment (toryhaavik) [reply ibis/pyspark/compiler.py]: i think just an issue here in Github is fine

Comment (hjoo) [reply ibis/pyspark/compiler.py]: https://github.com/ibis-project/ibis/issues/1957

**Comment by hjoo** [comment]
Most recent push is a rebase on top of master.

File: ibis/pyspark/tests/test_basic.py [hunk 2]
```diff
@@ -174,3 +174,28 @@ def test_join(client):
     )
 
     tm.assert_frame_equal(result.toPandas(), expected.toPandas())
+
+
+def test_filter(client):
```
Comment (xmnlab) [inline ibis/pyspark/tests/test_basic.py]: I think you should use pytest.mark.parametrize here. you can take a look here to have an idea:  https://github.com/ibis-project/ibis/blob/master/ibis/tests/all/test_numeric.py#L15

Comment (hjoo) [reply ibis/pyspark/tests/test_basic.py]: Agreed, good idea.

**Comment by toryhaavik** [comment]
This LGTM, @hjoo please just rebase on top of master, and i'll merge on green

**Comment by hjoo** [comment]
Rebase complete

**Pull request merged**